update/v1alpha1/types_update_status.go

```diff
@@ -0,0 +1,394 @@
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// UpdateStatus reports status for in-progress cluster version updates
+//
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +openshift:compatibility-gen:level=4
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:resource:path=updatestatuses,scope=Cluster
+// +openshift:api-approved.openshift.io=https://github.com/openshift/api/pull/2012
+// +openshift:file-pattern=cvoRunLevel=0000_00,operatorName=cluster-version-operator,operatorOrdering=02
+// +openshift:enable:FeatureGate=UpgradeStatus
+// +kubebuilder:metadata:annotations="description=Provides health and status information about OpenShift cluster updates."
+// +kubebuilder:metadata:annotations="displayName=UpdateStatuses"
+type UpdateStatus struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec is empty for now, UpdateStatus is purely status-reporting API. In the future spec may be used to hold
+	// configuration to drive what information is surfaced and how
+	// +required
+	Spec UpdateStatusSpec `json:"spec"`
+	// status exposes the health and status of the ongoing cluster update
+	// +optional
+	Status UpdateStatusStatus `json:"status"`
+}
+
+// UpdateStatusSpec is empty for now, UpdateStatus is purely status-reporting API. In the future spec may be used
+// to hold configuration to drive what information is surfaced and how
+type UpdateStatusSpec struct {
+}
+
+// +k8s:deepcopy-gen=true
+
+// UpdateStatusStatus is the API about in-progress updates. It aggregates and summarizes UpdateInsights produced by
+// update informers
+type UpdateStatusStatus struct {
+	// conditions provide details about the controller operational matters
```

Comment:
Do we know the kinds of conditions that might exist in here?

Reply:
Not 100% sure yet. Right now USC has three sub-controllers, I expected each of them maintain some kind of "alive and well" condition but I do not have a consumer or a use case for it yet.

Reply:
A TODO in here to update the list of valid conditions once we've cemented them would be good